tests/test_delb_plugin.py

```diff
@@ -0,0 +1,66 @@
+import pytest
+
+
+from delb import Document
+from snakesist import ExistClient
+
+
+@pytest.mark.usefixtures("db")
+def test_delete_document():
+    url = "existdb://admin:@localhost:8080/test_collection/test_document.xml"
+    document = Document(url)
+    document.existdb_delete()
+
+
+@pytest.mark.usefixtures("db")
+def test_load_document_from_url():
+    url = "existdb://admin:@localhost:8080/test_collection/test_document.xml"
+    document = Document(url)
+
+    assert document.config.source_url == url
+    assert isinstance(document.config.existdb.abs_id, str)
+    assert isinstance(document.config.existdb.client, ExistClient)
+    assert document.existdb_collection == "/test_collection/"
+    assert document.existdb_filename == "test_document.xml"
+
+
+@pytest.mark.usefixtures("db")
+def test_load_document_with_client():
+    exist_client = ExistClient(host="localhost", port=8080, user="admin", password="")
+    document = Document(
+        "/test_collection/test_document.xml", existdb_client=exist_client
```

Comment:
Frank came up with this in #15:

> this might result in ambiguities. let's say the document would be initialized with an additional client for another storage backend that could also make sense of `"/test_collection/test_document.xml"`.

but Theo was wondering:

> I'm probably not picturing this correctly, but I'm not sure I understand where this would be ambiguous. What would be the expected behavior from a document with two storage backends and where would it go wrong in this case?

Reply:
storing wouldn't be affected as i propose one (or more) distinct store methods per backend (`existdb_store`here), but the ambiguity could rise when two different backends are explicitly configured that use the same semantics to address documents within their storage, like this:

```python
existdb_client = snakesist.ExistClient(…)
other_client = TheBackendToTheXMLDatabaseThatSolvesJustOneProblem(…)

document = Document(
    "/test_collection/test_document.xml",
    existdb_client=existdb_client,
    other_client=other_client,
)
```

the notation in use is reasonably simple enough to be used in different contexts, so the scenario isn't too far fetched imo. with the current approach that the first loader that succeeds wins, this scenario would lead to an unresolvable conflict.

Reply:
if the problem is the arbitrary order of "first loader that succeeds", would taking the first specified backend as the source of the initial load be a solution?

Reply:
i created this issue about it: https://github.com/funkyfuture/delb/issues/9